src/main/java/github/scarsz/discordsrv/listeners/DiscordChatListener.java

```diff
@@ -105,17 +106,35 @@ public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
             DiscordSRV.debug("DiscordGuildMessagePreProcessEvent was cancelled, message send aborted");
             return;
         }
-
+        
+        List<Role> selectedRoles = new ArrayList<>();
+        // if we have a whitelist in the config
+        if (DiscordSRV.config().getBoolean("DiscordChatChannelRolesSelectionAsWhitelist")) {
+            selectedRoles = event.getMember().getRoles().stream()
+                               .filter(role -> DiscordSRV.config().getStringList("DiscordChatChannelRolesSelection").contains(DiscordUtil.getRoleName(role)))
+                               .collect(Collectors.toList());
+        } else { // if we have a blacklist in the settings
+            selectedRoles = event.getMember().getRoles().stream()
+                               .filter(role -> !DiscordSRV.config().getStringList("DiscordChatChannelRolesSelection").contains(DiscordUtil.getRoleName(role)))
```

Comment:
for the sake of preformance, it would be better to assign a variable, rather than querying the list on every role (also applies above, on whitelist)

Reply:
I assigned the config list to a variable, I hope the name is alright